Hw#8 clean swift

StasonLV opened the pull request.

StasonLV committed:
started

StasonLV committed:
Продолжаю структурировать

StasonLV committed:
Продолжаю структурировать

StasonLV committed:
NoteListScene

StasonLV committed:
удалено лишнее

StasonLV committed:
Добавил сцену заметки. реализовал переход между сценами

StasonLV committed:
добавлен роутинг, реализован не до конца

StasonLV committed:
ячейка настраивается, осталось доработать роутинг и удаление

StasonLV committed:
ДЗ готово

StasonLV committed:
Поправил удаление заметок(теперь в цикле)

StasonLV committed:
исправлены замечания? вынесен метод удаления в цикл, детали и создания заметки это простая вьюха, изменены методы роутера

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListViewController.swift:
```diff
@@ -1,14 +1,15 @@
-final class ListViewController: UIViewController {
+final class NoteListViewController: UIViewController {
     // MARK: - константы
+    var notes = [NoteListCleanModel.FetchData.ViewModel]()
```
private, общаемся только методами в данном случае через протоколы

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListViewController.swift:
```diff
@@ -35,10 +36,11 @@ final class ListViewController: UIViewController {
-    var notes = [NoteModel]()
-    let worker: WorkerType = Worker()
-    let loading = LoadingViewController()
+    var selected = [IndexPath]()
```
private

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListViewController.swift:
```diff
@@ -35,10 +36,11 @@ final class ListViewController: UIViewController {
-    let loading = LoadingViewController()
+    var selected = [IndexPath]()
+    private var notesForDeletion = [NoteListCleanModel.DeleteData.Request]()
+    let notesTable = UITableView(frame: .zero, style: .insetGrouped)
```
в протокол

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListViewController.swift:
```diff
@@ -76,23 +88,44 @@ final class ListViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
         setupNotesTable()
```
setupUI()

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListViewController.swift:
```diff
@@ -239,11 +175,11 @@ extension ListViewController: UITableViewDataSource, UITableViewDelegate, NotePr
         }
-        cell.userShareIcon.downloadImageFrom(urlString: notes[indexPath.row].userShareIcon ?? "")
         cell.delegate = self
+        cell.userShareIcon.downloadImageFrom(urlString: notes[indexPath.row].userShareIcon ?? "")
```
загрузку из вью вынести нужно, общались с тобой как это можно сделать

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListWorker.swift:
```diff
@@ -0,0 +1,45 @@
+            guard let data = data,
+                  let notes = try? JSONDecoder().decode([NoteListCleanModel.FetchData.Response].self, from: data)
+            else {
+                return
```
перед return должен быть completion, иначе будет баг когда не сможет декодировать

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListPresenter.swift:
```diff
@@ -0,0 +1,47 @@
+    }
+}
+
+extension Date {
```
в модель выноси, показывал как можно сделать в моделе приватным экстеншен

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListInteractor.swift:
```diff
@@ -0,0 +1,39 @@
+import UIKit
+
+final class NoteListInteractor: NoteListBusinessLogic, NoteListDataStore {
+    var note: NoteListCleanModel.FetchData.ViewModel?
```
интеракто не должен ничего знать о ViewModel это сущность view, и не вижу что бы она где то тут записывалась

bsciosedu commented on Trainee app/NotesCleanSwift/PresentationLayer/NoteListScene/NoteListRouter.swift:
```diff
@@ -0,0 +1,42 @@
+import UIKit
+
+final class NoteListRouter: NoteListRoutingLogic, NoteListDataPassing {
+    weak var viewController: NoteListViewController?
```
под протокол, а не наследование

StasonLV committed:
исправил обработку даты и нейминг

StasonLV committed:
убрал наследование контроллера, изменил метод работы роутера, табличка и массив дата сорса выделены в приватные

StasonLV committed:
удалены лишние файлы

StasonLV committed:
Вынес загрузку из вью

StasonLV committed:
пофиксил замечания
